study1/week1/merge-nodes-in-between-zeros.java

```diff
@@ -0,0 +1,30 @@
+/**
+ * Definition for singly-linked list.
+ * public class ListNode {
+ *     int val;
+ *     ListNode next;
+ *     ListNode() {}
+ *     ListNode(int val) { this.val = val; }
+ *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
+ * }
+ */
+class Solution {
+    public ListNode mergeNodes(ListNode head) {
+        //time complexity O(n^2) -> because of while loop inside while loop
```

Comment:
Can you try to think again? The core part for calculating time complexity for this algorithm is on how many nodes you visit. But the number of loop doesn't decide how many nodes you visit.
For instance -Merge sort can use a nested loop (or recursive) implementation, but that doesn't mean that it will take `O(n^2)`, but it's `O(nlogn)`.